src/common_swagger_api/schema/groups.clj

```diff
@@ -0,0 +1,112 @@
+(defn group [group-descriptor]
+  (assoc (base-group group-descriptor)
+
+    (s/optional-key :display_name)
+    (describe String (str "The displayable " group-descriptor " name"))
+
+    (s/optional-key :extension)
+    (describe String (str "The internal " group-descriptor " name extension"))
+
+    :id_index
+    (describe String "The sequential ID index number")
+
+    :id
+    (describe String (str "The " group-descriptor " ID"))))
+
+(defn group-update [group-descriptor]
+  (-> (base-group group-descriptor)
+      (->optional-param :name)
+      (dissoc :type)))
+
+(defn group-stub [group-descriptor]
+  (-> (group group-descriptor)
+      (->optional-param :name)
+      (->optional-param :type)
+      (->optional-param :id)
+      (->optional-param :id_index)))
+
+(defn group-detail [group-descriptor]
+  (let [group-schema (group group-descriptor)]
+    {(s/optional-key :attribute_names)
+     (describe [String] (str "Attribute names, not including the ones listed in the " group-descriptor " itself"))
+
+     (s/optional-key :attribute_values)
+     (describe [String] (str "Attribute values, not including the ones listed in the " group-descriptor " itself"))
+
+     (s/optional-key :composite_type)
+     (describe String (str "The type of composite " group-descriptor ", if applicable"))
+
+     :created_at
+     (describe Long (str "The date and time the " group-descriptor " was created (ms since epoch)"))
+
+     :created_by
+     (describe String (str "The ID of the subject who created the " group-descriptor))
+
+     :has_composite
+     (describe Boolean (str "True if this " group-descriptor " has a composite member"))
+
+     :is_composite_factor
+     (describe Boolean (str "True if this " group-descriptor " is a composite member of another group"))
+
+     (s/optional-key :left_group)
+     (describe group-schema (str "The left " group-descriptor " if this group is a composite"))
+
+     (s/optional-key :modified_at)
+     (describe Long (str "The date and time the " group-descriptor " was last modified (ms since epoch)"))
+
+     (s/optional-key :modified_by)
+     (describe String (str "The ID of the subject who last modified the " group-descriptor))
+
+     (s/optional-key :right_group)
+     (describe group-schema (str "The right " group-descriptor " if this group is a composite"))
+
+     (s/optional-key :type_names)
+     (describe [String] (str "The types associated with this " group-descriptor))}))
+
+(defn group-with-detail [group-descriptor]
+  (assoc (group group-descriptor)
+    (s/optional-key :detail)
+    (describe (group-detail group-descriptor) (str "Detailed information about the " group-descriptor))))
+
+(defn group-list [group-descriptor plural-group-descriptor]
+  {:groups (describe [(group group-descriptor)] (str "The list of " plural-group-descriptor " in the result set"))})
+
+(defn group-members [group-descriptor]
+  {:members (describe [subjects/Subject] (str "The list of " group-descriptor " members"))})
```

Comment:
The idea behind these functions is that the service that is using these schemas will call them and pass the result to `schema.core/defschema` to actually define the schema, for example:

```clojure
(s/defschema BaseGroup (group-schema/base-group "group"))
(s/defschema Group (group-schema/group "group"))
(s/defschema GroupUpdate (group-schema/group-update "group"))
(s/defschema GroupStub (group-schema/group-stub "group"))
(s/defschema GroupWithDetail (group-schema/group-with-detail "group"))
(s/defschema GroupList (group-schema/group-list "group" "groups"))
(s/defschema GroupMembers (group-schema/group-members "group"))
(def GroupMembersUpdate group-schema/GroupMembersUpdate)
(def GroupMembersUpdateResponse group-schema/GroupMembersUpdateResponse)
```